COM/oslinux.cpp

```diff
@@ -367,13 +367,13 @@ static QString autoconfPath;
 */
 static bool isGoodKernelDir( const QString &kDir, QString &errMsg )
 {
-	// check if it's really a linux kernel include dir...
-	QFileInfo fi1( kDir + "/include/linux/kernel.h" );
+	// check if it's really a configured linux kernel dir...
+	QFileInfo fi1( kDir + "/Module.symvers" );
```

Comment:
According https://www.kernel.org/doc/html/latest/kbuild/modules.html?highlight=modules%20symvers the Module.symvers file is only available after performing a full kernel build.
Is this modification compatible to all MDIS supported linux distros? It should be possible to build MDIS without having to manually build a kernel first. Or does Module.symvers exists after just installing the kernel sources?

Reply:
This seems correct. I did not considered using the Linux sources with just the minimum configuration to compile kernel modules. It seems indeed that `make modules_prepare` does not create this Module.symvers. 

I'll double check what we can use that is simple enough to just say "ok this looks like a linux directory" that is compatible with the way Debian splits its headers package. Probably the way to fix this is to remove this initial check as there is a second check in this `isGoodKernelDir() ` that checks for `autoconf.h` in a `include/linux` directory.

Reply:
This one seems like an interesting candidate:
```
include/config/kernel.release
```

Reply:
OK I updated this patches so that include/config/kernel.release . This file gets created when calling `make modules_prepare` and it exists in the linux headers packages ready to compile. I checked the headers packages in Ubuntu 22, Debian 10 and CentOS. I also checked with the linux sources repo and did this:

```
jorge@jorge-Latitude-5530:~/Work/MDIS/linux$ cat include/config/kernel.release
cat: include/config/kernel.release: No such file or directory
jorge@jorge-Latitude-5530:~/Work/MDIS/linux$ make x86_64_defconfig
#
# No change to .config
#
jorge@jorge-Latitude-5530:~/Work/MDIS/linux$ cat include/config/kernel.release
cat: include/config/kernel.release: No such file or directory
jorge@jorge-Latitude-5530:~/Work/MDIS/linux$ make modules_prepare
  SYSHDR  arch/x86/include/generated/uapi/asm/unistd_32.h
  SYSHDR  arch/x86/include/generated/uapi/asm/unistd_64.h
.....
  CC      /home/jorge/Work/MDIS/linux/tools/objtool/str_error_r.o
  CC      /home/jorge/Work/MDIS/linux/tools/objtool/librbtree.o
  LD      /home/jorge/Work/MDIS/linux/tools/objtool/objtool-in.o
  LINK    /home/jorge/Work/MDIS/linux/tools/objtool/objtool
  LDS     scripts/module.lds
jorge@jorge-Latitude-5530:~/Work/MDIS/linux$ cat include/config/kernel.release
5.15.0
```

Reply:
I have tested it in CentOS 8 and it works properly.